forward/forwarder.go

```diff
@@ -186,6 +192,9 @@ func SetForwardedHeader(ctx thrift.Context) thrift.Context {
 // the message. When a message has already been forwarded by ringpop the
 // forwarding logic should not be used to prevent unbound forwarding.
 func HasForwardedHeader(ctx thrift.Context) bool {
-	_, ok := ctx.Headers()[forwardedHeaderName]
+	_, ok := ctx.Headers()[ForwardedHeaderName]
+	if ok {
+		delete(ctx.Headers(), ForwardedHeaderName)
```

Comment:
I'm not sure if we want the delete as a side-effect of a call to `HasForwardedHeader`.
Maybe change the function name to `DeleteForwardedHeader` and return true if deletion was successful (=header was present)?